modules/account-lib/src/coin/algo/keyRegistrationBuilder.ts

```diff
@@ -248,12 +274,26 @@ export class KeyRegistrationBuilder extends TransactionBuilder {
     super.validateTransaction(transaction);
     if (this.isOnlineKeyRegAccountLibTransaction()) {
       // invalid offline will reach here
-      this.validateFields(this._voteKey, this._selectionKey, this._voteFirst, this._voteLast, this._voteKeyDilution);
+      this.validateFields(
+        this._voteKey,
+        this._selectionKey,
+        this._voteFirst,
+        this._voteLast,
+        this._voteKeyDilution,
+        this._stateProofKey,
+      );
     } else {
       // offline or nonparticipation transaction
-      if (this._voteKey || this._selectionKey || this._voteFirst || this._voteLast || this._voteKeyDilution) {
+      if (
+        this._voteKey ||
+        this._selectionKey ||
+        this._voteFirst ||
+        this._voteLast ||
+        this._voteKeyDilution ||
+        this._stateProofKey
```

Comment:
```suggestion
isOfflineOrNonParticipation()
```

Reply:
The idea here is check if at least one of that parameters is not undefined. We are already using an "else" condition of "this.isOnlineKeyRegAccountLibTransaction()", and that method only check if that transaction is not a offline/nonparticipation.
I supose this could be most clear in this way. But, if you are not agree, I will change it.